table/block_based/block_based_filter_block.cc

```diff
@@ -230,27 +293,54 @@ bool BlockBasedFilterBlockReader::MayMatch(const Slice& entry,
 }
 
 size_t BlockBasedFilterBlockReader::ApproximateMemoryUsage() const {
-  return num_ * 4 + 5 + (offset_ - data_);
+  size_t usage = ApproximateFilterBlockMemoryUsage();
+#ifdef ROCKSDB_MALLOC_USABLE_SIZE
+  usage += malloc_usable_size((void*)this);
```

Comment:
Come to think of it, the pointer types in question can be implicitly converted to void pointers, and the only thing we need to cast away is the constness of the this pointer (since these are const methods). Changed to const_cast.